### ISSUE TITLE
Add support for creating Streams from Iterators

### ISSUE THREAD

**Issue by pvande** [winterbe/streamjs#37]
Supporting Iterators neatly permits both the built-in Iterator types and ES6 Generator functions.

``` javascript
var iterator = function* generate() {
  yield 1;
  yield 3;
  yield 7;
  yield 15;
};

Stream(iterator()).map(function(x) { return (x - 1) / 2 }).toArray();  // => [ 0, 1, 3, 7 ]
```

There may also be a case for supporting objects conforming to an "Enumerable" protocol (effectively providing a convenience for constructing iterators), but that seems like a secondary concern.

**Comment by winterbe** [comment]
This is definitely on the roadmap along with streaming objects from ES6 sets and maps.

Though I have to figure out how to support the new stuff while retaining backward compatibility to ES5.

**Comment by winterbe** [comment]
This feature has today been released in version 1.6.0. Again, thanks for your help. :+1:
